### PR TITLE
new context on BrewingStand event

### PR DESCRIPTION
**Pull request by pobab**
request from https://discord.com/channels/315163488085475337/1136494737277321389

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java [hunk 1]
```diff
@@ -25,6 +30,8 @@ public class BrewsScriptEvent extends BukkitScriptEvent implements Listener {
     // @Context
     // <context.location> returns the LocationTag of the brewing stand.
     // <context.inventory> returns the InventoryTag of the brewing stand's contents.
+    // <context.fuel_level> returns an ElementTag of the brewing stand's fuel.
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java]: `ElementTag(Number)`, `fuel level`

File: plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java [hunk 2]
```diff
@@ -48,6 +55,16 @@ public ObjectTag getContext(String name) {
...
+            case "fuel_level": return new ElementTag(event.getFuelLevel());
+            case "result":
+                ListTag results = new ListTag();
+                for (ItemStack itemStack : event.getResults()){
+                    if (itemStack == null){
+                        results.addObject(new ItemTag(Material.AIR));
+                    }
+                    results.addObject(new ItemTag(itemStack));
+                }
+                return results;
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java]: Can use the ListTag convertor constructor - `new ListTag(getResults, ItemTag::new)`

**Comment by pobab** [comment]
done

File: plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java [hunk 3]
```diff
@@ -1,12 +1,17 @@
 package com.denizenscript.denizen.events.block;
 
 import com.denizenscript.denizen.objects.InventoryTag;
+import com.denizenscript.denizen.objects.ItemTag;
 import com.denizenscript.denizen.objects.LocationTag;
 import com.denizenscript.denizen.events.BukkitScriptEvent;
 import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.objects.core.ListTag;
+import org.bukkit.Material;
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java]: You seem to have redundant imports now - can remove these manually or use the IDE's automatic import cleanup (`Ctrl + Alt + O`)

Comment (pobab) [reply plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java]: i never knew that shortcut existed, thanks aya!

**Comment by tal5** [comment]
Looks good so far, but not merged yet as [a determination](https://discord.com/channels/315163488085475337/1011496047811506227/1140275063338315946) is being added.

File: plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java [hunk 4]
```diff
@@ -25,11 +28,22 @@ public class BrewsScriptEvent extends BukkitScriptEvent implements Listener {
...
+    // <context.result> returns a ListTag(ItemTag) of the brewing stand's result.
+    //
+    // @Determine
+    // "RESULT:<ListTag(ItemTag)>" to change the item that is brewed.
     //
     // -->
 
     public BrewsScriptEvent() {
         registerCouldMatcher("brewing stand brews");
+        this.<BrewsScriptEvent, ListTag>registerDetermination("result" ,ListTag.class, (evt, context, listTag) -> {
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java]: spacing is wonky here. And. uh. throughout this block.

File: plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java [hunk 5]
```diff
@@ -25,11 +28,22 @@ public class BrewsScriptEvent extends BukkitScriptEvent implements Listener {
...
     // -->
 
     public BrewsScriptEvent() {
         registerCouldMatcher("brewing stand brews");
+        this.<BrewsScriptEvent, ListTag>registerDetermination("result", ListTag.class, (evt, context, listTag) -> {
+            evt.event.getResults().clear();
+            for (ItemTag item : listTag.filter(ItemTag.class, context)){
+                evt.event.getResults().add(item.getItemStack());
+            }
+        } );
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java]: Redundant space here

File: plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java [hunk 6]
```diff
@@ -25,11 +28,22 @@ public class BrewsScriptEvent extends BukkitScriptEvent implements Listener {
...
+    // <context.result> returns a ListTag(ItemTag) of the brewing stand's result.
+    //
+    // @Determine
+    // "RESULT:<ListTag(ItemTag)>" to change the item that is brewed.
     //
     // -->
 
     public BrewsScriptEvent() {
         registerCouldMatcher("brewing stand brews");
+        this.<BrewsScriptEvent, ListTag>registerDetermination("result", ListTag.class, (evt, context, listTag) -> {
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java]: Should probably have a more descriptive name rather then just `listTag` (e.g. `result`)

File: plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java [hunk 7]
```diff
@@ -25,11 +28,22 @@ public class BrewsScriptEvent extends BukkitScriptEvent implements Listener {
     // @Context
     // <context.location> returns the LocationTag of the brewing stand.
     // <context.inventory> returns the InventoryTag of the brewing stand's contents.
+    // <context.fuel_level> returns an ElementTag(Number) of the brewing stand's fuel level.
+    // <context.result> returns a ListTag(ItemTag) of the brewing stand's result.
+    //
+    // @Determine
+    // "RESULT:<ListTag(ItemTag)>" to change the item that is brewed.
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java]: It's multiple items, maybe something like `to change the items that will be brewed`

File: plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java [hunk 8]
```diff
@@ -25,11 +28,22 @@ public class BrewsScriptEvent extends BukkitScriptEvent implements Listener {
     // @Context
     // <context.location> returns the LocationTag of the brewing stand.
     // <context.inventory> returns the InventoryTag of the brewing stand's contents.
+    // <context.fuel_level> returns an ElementTag(Number) of the brewing stand's fuel level.
+    // <context.result> returns a ListTag(ItemTag) of the brewing stand's result.
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java]: Would be nice to have the context's meta match the determination's meta, e.g. `the items that will be brewed`/whatever you make the determination meta

File: plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java [hunk 9]
```diff
@@ -25,11 +28,22 @@ public class BrewsScriptEvent extends BukkitScriptEvent implements Listener {
...
+    // @Determine
+    // "RESULT:<ListTag(ItemTag)>" to change the items that are brewed.
     //
     // -->
 
     public BrewsScriptEvent() {
         registerCouldMatcher("brewing stand brews");
+        this.<BrewsScriptEvent, ListTag>registerDetermination("result", ListTag.class, (evt, context, result) -> {
+            evt.event.getResults().clear();
+            for (ItemTag item : result.filter(ItemTag.class, context)){
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/events/block/BrewsScriptEvent.java]: spacing still weird here

**Pull request merged**